python_modules/libraries/dagster-dg-cli/dagster_dg_cli/cli/scaffold/branch/ai.py

```diff
@@ -128,18 +131,81 @@ def additional_allowed_tools(cls) -> list[str]:
 
 
 def get_branch_name_and_pr_title_from_prompt(
-    dg_context: DgContext, user_input: str, input_type: type[InputType]
+    dg_context: DgContext,
+    user_input: str,
+    input_type: type[InputType],
+    diagnostics: ClaudeDiagnosticsService,
 ) -> tuple[str, str]:
     """Invokes Claude under the hood to generate a reasonable, valid
     git branch name and pull request title based on the user's stated goal.
     """
+    context_str = input_type.get_context(user_input)
+    prompt = load_branch_name_prompt(context_str)
+    allowed_tools = input_type.additional_allowed_tools()
+
+    if diagnostics.level != "off":
```

Comment:
IIUC this conditional (and others like it) is redundant because a level check is already run in the service `log` method.

I guess it could save some non-trivial computation in some cases. Too bad there are no multiline lambdas, would be nice here.

If there are a lot of checks might be nice to have a `.enabled` property instead of the "off" check